### ISSUE TITLE
Update: Cross-Site Request Forgery Prevention Cheat Sheet

### ISSUE THREAD

**Issue by glasser** [OWASP/CheatSheetSeries#1011]
## What is missing or needs to be updated?

Cross-Site Request Forgery attacks occur because untrusted browser code can cause requests to be sent to a vulnerable server that are "special" in a way that the attacker could not send from their own machines.  Typically this "special"-ness is that they contain cookies from a targeted user's browser. Less commonly, this could mean they contain basic auth credentials or that the server makes authorization assumptions based on the client's network (eg, explicitly looking at the client's IP address to make decisions, or by putting the server on a private network and assuming that any client talking to this server is authorized).

The mitigations discussed in the cheat sheet all assume that these "special" request qualities must exist in your system. However, an entirely different approach to solving CSRF is to avoid giving special treatment based on cookies, basic auth, and network properties.  This approach may be appropriate for web servers that are primarily serving "API-style" traffic and which websites only communicate with via AJAX-style calls.

For example, a server that makes authentication decisions only based on a custom HTTP header (perhaps set from a value stored in browser `localStorage`) or a value in the body of a POST and not on cookies or network should be considered successfully defended against CSRF. Developers who have built servers with this property should not feel like they need to also set up token-based mitigation or (if #1010 is accepted) header-based mitigation.

## How should this be resolved?

A new top-level section should be added describing "don't care about cookies/etc" as an appropriate mitigation technique, with the drawback that it is not helpful if you need your endpoint to be accessible via HTML `<form>`s.

**Comment by mackowski** [comment]
Yes, that would be a good addition! Do you want to address it in PR?

**Comment by professorabhay** [comment]
I am interested to work on this.

**Comment by professorabhay** [comment]
Hey 👋 @mackowski
I am interested to work on this issue.

**Comment by glasser** [comment]
@professorabhay I haven't started working on this or #1010 (which might want to be just one rewrite that addresses both) and would be happy to review your attempt instead!

**Comment by professorabhay** [comment]
Hello @mackowski, Yes, I am interested to work on both these issues. But Can you please explain it little bit more.....
I have to update that file -
`https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md `

According to problem and solution mentioned.

I am new in these type contributions but I can try hard to do.

**Comment by mackowski** [comment]
Exactly you need to update Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md and create a Pull Request for us to review. If you will have any questions just ask :)

**Comment by mackowski** [comment]
Hey @Yzabhellnelton go ahead and work on this :) we will help you and review the PR :)

**Comment by ouch3n** [comment]
Hi, are you still working on it ? If not, I would like to work on this issue if possible. Please

**Comment by mackowski** [comment]
@ouch3n I will assign this to you. Thanks!

**Comment by mackowski** [comment]
@ouch3n any updates on this from your side?

**Comment by ouch3n** [comment]
@mackowski Sorry, I was busy this period, by the end of the week I plan to do a PR! - WIP

**Comment by mackowski** [comment]
@ouch3n any updates on this?

**Comment by jmanico** [comment]
This is fairly old so I am going to politely close it out, but please re-open if interest is still there.

**Comment by glasser** [comment]
This does appear to still not be really covered in the page, for what it's worth. But I also haven't tried it fix it myself in 3+ years.

**Comment by jmanico** [comment]
Would you care to give us a brief PR to mention this?
